test/test_mimic_explainer.py

```diff
@@ -492,17 +513,35 @@ def test_linear_explainable_model_classification(self, mimic_explainer):
         cat_transformations = [([cat_feature_name], encoder) for cat_feature_name, encoder in
                                zip(cat_feature_names, cat_feature_encoders)]
 
-        explainer = mimic_explainer(model=model,
-                                    initialization_examples=data_x,
-                                    explainable_model=LinearExplainableModel,
-                                    explainable_model_args={'sparse_data': True},
-                                    augment_data=False,
-                                    features=cat_feature_names,
-                                    classes=['0', '1'],
-                                    transformations=cat_transformations,
-                                    model_task=ModelTask.Classification)
-        global_explanation = explainer.explain_global(evaluation_examples=data_x)
+        if raw_feature_transformations:
+            explainer = mimic_explainer(model=model,
+                                        initialization_examples=data_x,
+                                        explainable_model=LinearExplainableModel,
+                                        explainable_model_args={'sparse_data': True},
+                                        augment_data=False,
+                                        features=cat_feature_names,
+                                        classes=classes,
+                                        transformations=cat_transformations,
+                                        model_task=ModelTask.Classification)
+            global_explanation = explainer.explain_global(evaluation_examples=data_x)
+        else:
+            explainer = mimic_explainer(model=model,
+                                        initialization_examples=encoded_cat_features,
+                                        explainable_model=LinearExplainableModel,
+                                        explainable_model_args={'sparse_data': True},
+                                        augment_data=False,
+                                        classes=classes,
+                                        model_task=ModelTask.Classification)
+            global_explanation = explainer.explain_global(evaluation_examples=encoded_cat_features)
+
         assert global_explanation.method == LINEAR_METHOD
+        if if_multiclass:
+            if raw_feature_transformations:
+                predictions_surrogate_model = explainer._get_surrogate_model_predictions(data_x)
+                assert classes == np.unique(predictions_surrogate_model).tolist()
```

Comment:
nit: move this assert under if_multiclass and remove the duplicate assert below on line 544

Reply:
Done